### PR TITLE
New post: Intel and Hugging Face Partner to Democratize Machine Learning Hardware Acceleration

### PR DESCRIPTION
**Pull request by juliensimon**
Feel free to edit and publish, @feconroses :)

### CODE REVIEW

File: intel.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+```
+from evaluate import evaluator
+
+def eval_func(model): 
+    eval = evaluator("text-classification") 
+    results = eval.compute(
+        model_or_pipeline=model, tokenizer=tokenizer, 
+        data=eval_dataset, metric=evaluate.load("accuracy"), 
+    label_column="labels", label_mapping=model.config.label2id
+    )
```
Comment (echarlaix) [inline intel.md:90]:
```suggestion
    results = eval.compute(
        model_or_pipeline=model,
        tokenizer=tokenizer,
        data=eval_dataset,
        metric=evaluate.load("accuracy"),
        label_column="labels",
        label_mapping=model.config.label2id,
    )
```

File: intel.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+```
+from optimum.intel.neural_compressor import IncOptimizer, IncQuantizer
+from optimum.intel.neural_compressor.configuration import IncQuantizationConfig
+
+quantization_config = IncQuantizationConfig.from_pretrained(
+    config_name_or_path="juliensimon/distilbert-amazon-shoe-reviews",
+    config_file_name="quantize.yml"
+)
+inc_quantizer = IncQuantizer(model, quantization_config, eval_func=eval_func)
```
Comment (echarlaix) [inline intel.md:114]:
```suggestion
inc_quantizer = IncQuantizer(quantization_config, eval_func=eval_func)
```

File: intel.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+```
+
+We can now launch the quantization job.
+
+```
+inc_optimizer = IncOptimizer(model, quantizer=inc_quantizer)
+inc_model = inc_optimizer.fit()
+```
+
+The log tells us that Optimum Intel has quantized 39 ```Linear``` operators.
```
Comment (echarlaix) [inline intel.md:124]:
```suggestion
The log tells us that Optimum Intel has quantized 38 ```Linear``` operators.
```

File: intel.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+## Case study: Quantizing DistilBERT with Optimum Intel
+
+In this example, we will run post-training quantization on a DistilBERT model fine-tuned for classification. Quantization is a process that shrinks memory and compute requirements by reducing the bit width of model parameters. For example, you can often replace 32-bit floating-point parameters with 8-bit integers at the expense of a small drop in prediction accuracy.
+
+We have already fine-tuned the original model to classify product reviews for shoes according to their star rating (from 1 to 5 stars). You can view this [model](https://huggingface.co/juliensimon/distilbert-amazon-shoe-reviews) and its [quantized](https://huggingface.co/juliensimon/distilbert-amazon-shoe-reviews-quantized?) version on the Hugging Face hub. You can also test the original model in this [Space](https://huggingface.co/spaces/juliensimon/amazon-shoe-reviews-spaces). 
+
+Let’s get started! All code is available in this [notebook](https://gitlab.com/juliensimon/huggingface-demos/-/blob/main/amazon-shoes/03_optimize_inc_quantize.ipynb).
```
Comment (echarlaix) [inline intel.md:58]: The notebook code is for optimum-intel version < 1.2.3, can I help update it ?

File: intel.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+---
+
+
+
+<h1>Intel and Hugging Face Partner to Democratize Machine Learning Hardware Acceleration</h1>
+
+
+
+<div class="blog-metadata">
+    <small>Published May 16, 2022.</small>
```
Comment (stefan-it) [inline intel.md:13]: Needs an update here :thinking:

File: intel.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+[INFO] +--------------------+-----------+---------------+------------------+
+```
+
+Accuracy didn’t drop, and Optimum Intel stopped the quantization job after the first step. Had accuracy dropped more than the allowed 3%, Optimum Intel would have tried to quantize different parts of the models until it would have reached an acceptable drop, or the maximum number of trials.
+
+Finally, we save the model and its configuration file to local storage.
+
+```
+model_dir = "./model_inc"
+inc_model.model.save_pretrained(model_dir)
```
Comment (echarlaix) [inline intel.md:200]:
```suggestion
inc_optimizer.save_pretrained(model_dir)
```

File: intel.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+```
+
+Once we’ve created a new [repository](https://huggingface.co/juliensimon/distilbert-amazon-shoe-reviews-quantized) on the Hugging Face hub and pushed the quantized model to it, we can load it in the usual way.
+
+```
+from optimum.intel.neural_compressor.quantization import IncQuantizedModelForSequenceClassification
+
+inc_model = IncQuantizedModelForSequenceClassification.from_pretrained(
+    "juliensimon/distilbert-amazon-shoe-reviews-quantized")
```
Comment (echarlaix) [inline intel.md:209]:
```suggestion
inc_model = IncQuantizedModelForSequenceClassification.from_pretrained(
    "juliensimon/distilbert-amazon-shoe-reviews-quantized"
)
```

**Review by echarlaix — COMMENTED** [review]
The blog is super clear and well explained, thanks for the amazing work @juliensimon

File: intel.md (hunk 4 above)
Comment (juliensimon) [reply intel.md:58]: updated.

File: intel.md [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+    (lin1): DynamicQuantizedLinear(in_features=768, out_features=3072, dtype=torch.qint8, qscheme=torch.per_channel_affine)
+    (lin2): DynamicQuantizedLinear(in_features=3072, out_features=768, dtype=torch.qint8, qscheme=torch.per_channel_affine)
+  )
+  (output_layer_norm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)
+)
+```
+
+Very well, but how does this impact accuracy and prediction time?
+
+Before and after each quantization step, Optimum Intel runs the evaluation function on the current model. Interestingly, the accuracy of the quantized model is now a bit higher (```0.6906```) than the original model (```0.6893```). Likewise, we see that the quantized model predicts the evaluation set 13% faster than the original model. Not bad for a few lines of code!
```
[the remark this answers is not in the document]
Comment (juliensimon) [reply intel.md:185]: Interesting, I didn't know that! I'll add it to the post.

**Review by echarlaix — APPROVED** [review]
